tests/framework/extensions/nodes/nodes.go

```diff
@@ -0,0 +1,123 @@
+package nodes
```

Comment:
Just a reminder that as this is an extension, this needs to go to `rancher/shepherd`. So, whenever we're given the green light, this needs to stripped from this PR and put in a separate PR in that repo.